earthkit/geo/distance.py

```diff
@@ -139,45 +146,59 @@ def nearest_point_haversine(ref_points, points):
     res_index = []
     res_distance = []
     for lat, lon in ref_points.T:
-        distance = haversine_distance((lat, lon), points).flatten()
+        distance = haversine_distance((lat, lon), points, radius=1).flatten()
         index = np.nanargmin(distance)
         index = index[0] if isinstance(index, np.ndarray) else index
         res_index.append(index)
         res_distance.append(distance[index])
-    return (np.array(res_index), np.array(res_distance))
+    return (np.array(res_index), np.array(res_distance) * radius)
 
 
-def ll_to_xyz(lat, lon):
+def _ll_to_xyz(lat, lon, radius=constants.R_earth):
     lat = np.asarray(lat)
     lon = np.asarray(lon)
     lat = np.radians(lat)
     lon = np.radians(lon)
-    x = constants.R_earth * np.cos(lat) * np.cos(lon)
-    y = constants.R_earth * np.cos(lat) * np.sin(lon)
-    z = constants.R_earth * np.sin(lat)
-    return x, y, z
+    x = np.cos(lat) * np.cos(lon)
+    y = np.cos(lat) * np.sin(lon)
+    z = np.sin(lat)
+
+    if radius == 1:
+        return x, y, z
+    else:
+        return radius * x, radius * y, radius * z
 
 
-def cordlength_to_arclength(chord_length):
+def _cordlength_to_arclength(chord_length, radius=constants.R_earth):
```

Comment:
chord length; I would prefer "chord to arc" but of course I understand "chord length to arc length"